### DISCUSSION TITLE
The npmjs website doesn’t show the TypeScript badge for packages that use simple package exports

### DISCUSSION THREAD

**Discussion by remcohaszing** [community/community#134258] [npm]
[`codemirror-languageservice`](https://www.npmjs.com/package/codemirror-languageservice/v/0.1.0) is fully typed, but it doesn’t have the TypeScript badge on the package page on npmjs.com.

See https://arethetypeswrong.github.io/?p=codemirror-languageservice@0.1.0 for details. I decided to not support the `node10` module resolution, as people should no longer use that in their projects.

**Comment by thescientist13** [comment]
Looks like it's required to have a "top" level `types` or `typings` field in _package.json_ for this take effect - https://github.blog/changelog/2020-12-16-npm-displays-packages-with-bundled-typescript-declarations/

I had `types` defined in the `exports` field, but the badge doesn't show (which would be nice)
```json
{
  "exports": {
    ".": {
      "types": "./src/types/index.d.ts",
      "import": "./src/index.js"
    },
  }
}
```

 So will try adding the `types` field and see what that does 🤞

**Comment by R-Bower** [comment]
I use a stub file for this.  Kinda sucks that there's no other way.

https://github.com/qualcomm/qualcomm-ui/blob/main/packages/frameworks/react/package.json#L16

```ts
/**
 * @packageDocumentation
 * This file is here to ensure that npm shows the TypeScript badge.
 * This package uses secondary entrypoints only.
 * Import from 'pkg/subpath1', 'pkg/subpath2', etc.
 */
export {}
```

<img width="304" height="51" alt="image" src="https://github.com/user-attachments/assets/7fbbdc27-a8e0-4552-b8b1-90de6b809470" />

**Comment by assanftako-tech** [comment]
Uploading 9f0c30220764f83c2b5ad2c14c423164.mp4…

**Comment by assanftako-tech** [comment]
Bonsoir
